pkg/config/runtime_test.go

```diff
@@ -37,14 +42,14 @@ spec:
   nodeConfig:
     metadata:
       name: k0s
-  pid: -1
 `)
 	require.NoError(t, os.WriteFile(rtConfigPath, content, 0644))
 
 	// try to load runtime config and check if it returns an error
 	spec, err := LoadRuntimeConfig(rtConfigPath)
-	assert.Nil(t, spec)
-	assert.ErrorIs(t, err, ErrK0sNotRunning)
+	require.NoError(t, err)
+	assert.NotNil(t, spec)
+	require.NoError(t, spec.Cleanup())
```

Comment:
Isn't this reverting how the test is working? I think the idea was to actually check that ErrNotRunning is returned.

Reply:
Yes. It changes how the test is working because it changes how `LoadRuntimeConfig` works. With this change we *only* load the runtime config and no longer perform any checks if k0s is running or not. It return failures if it fails to load the runtime config and returns success if it successds to read it, regardless if k0s is running or not.

The test for other running k0s instances moved to NewRuntimeConfig and only happens from there.

The only reference to `LoadRuntimeConfig` (except the test) is in https://github.com/k0sproject/k0s/blob/cb157750dac18fac41107ab53531c897ff820175/pkg/config/cli.go#L305

I didn't think it was worth spending time on, but I can add it back if you want.

Reply:
I added a `Locked()` function to restore original behavior.